donate/payments/views.py

```diff
@@ -798,15 +804,22 @@ def form_valid(self, form, send_data_to_basket=True):
                 )
 
         elif send_data_to_basket:
-            data['source_url'] = self.request.build_absolute_uri()
-            data['lang'] = self.request.LANGUAGE_CODE
-            queue.enqueue(send_newsletter_subscription_to_basket, data)
-            queue_ga_event(self.request, ['send', 'event', {
-                    'eventCategory': 'Signup',
-                    'eventAction': 'Submitted the Form',
-                    'eventLabel': 'Email',
-                }
-            ])
+            # We are switching from sending data to SQS, to sending data to Basket instead.
+            # During the switch phase, we're supporting both with an environment variable that
+            # determines where it should go.
+            if settings.DONATE_NEWSLETTER_SUBSCRIBE_METHOD == 'BASKET':
+                basket.subscribe(data['email'], 'mozilla-foundation', lang=self.request.LANGUAGE_CODE)
+            elif settings.DONATE_NEWSLETTER_SUBSCRIBE_METHOD == "SQS":
+                data['source_url'] = self.request.build_absolute_uri()
+                data['lang'] = self.request.LANGUAGE_CODE
+                queue.enqueue(send_newsletter_subscription_to_basket, data)
+
+        queue_ga_event(self.request, ['send', 'event', {
```

Comment:
I believe the original behaviour was to only send the event if send_data_to_basket is truthy. Based on the indent level of this function call, it looks like it's now outside the `elif send_data_to_basket` scope. Is this intended?

Reply:
It was, but you raise a good point. It shouldn't be. I had overlooked that. Thanks for spotting this!